.github/workflows/lint_code.yaml

```diff
@@ -84,7 +84,7 @@ jobs:
           done
           rm expanded.tmp
       - name: Lint YAML
-        uses: karancode/yamllint-github-action@dd59165b84d90d37fc919c3c7dd84c7e37cd6bfb  # this is 2.0.0
+        uses: karancode/yamllint-github-action@2.1.1
```

Comment:
Are we comfortable pinning to a version not a commit here? The original reason for doing so was to stop arbitrary 3rd party code from running in this repo without review. Note that the `2.1.1` tag can move while the commit hash can't.

Reply:
I am. I've left the create PR action as a commit hash, however. I remember our justification being that we should be careful with actions that can make, or propose, changes to the code.

This job doesn't have the permissions to make changes.

Reply:
I'm personally not worried about any job that makes changes which need to be reviewed. Is there any way that a GitHub action can make a change without going through a PR? If not, I think we can probably make all of these actions use tags.

Reply:
Actually, I think I was wrong

- https://docs.github.com/en/actions/security-guides/automatic-token-authentication#about-the-github_token-secret
- https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions

In any case, you are right that CODEOWNERS and branch protection rules mean the most an action could do is open a PR, push to an unprotected branch. We should still be reviewing all changes even if they aren't made by a person.

I would be pretty comfortable using tags.